sdk/iot/core/CMakeLists.txt

```diff
@@ -15,8 +15,10 @@ add_library (
 target_include_directories (az_iot PUBLIC inc)
 target_include_directories (az_iot PRIVATE src)
 
-target_link_libraries(az_iot PUBLIC az_core)
+# Get only headers from core. Do not link. Customers is expected to link
+# with az_iot and az_core. This allows sharing az_core with more clients
+target_include_directories (az_iot PRIVATE ../../core/core/inc)
```

Comment:
This will likely be a problem if someone decides to try and use this when installed. The relative paths will not resolve. The `target_link_libraries` case can in theory work with installed libs or locally built if the libs are populated correctly. This also seems to break the CMake benefit of dependencies transitively associated to targets for ease of use by the user. What do you mean by
>allows sharing az_core with more clients

this should be fine at link time if you have a situation like this:
```
                  az_iot -> az_core
                / 
user_app ->
                \
                  az_identity -> az_core
```
There shouldn't be a duplication of code.

Reply:
The situation is when a customer would like to have different az_core versions.
Like, imagine today they build az_iot tiwh az_core (let's called v1)
Then, after some time (or years), the want to use az_keyvault that is linked with az_core v2.
If they try to link az_oit and az_kayvault, linker will found duplicated definitions from az_core v1 and az_core v2 right? .

I might be wrong, but, my understanding is that when we create a library like az_iot or az_storage and call target_link_libraries(az_core), for an static library, we would be taking all implementation from az_core (and not only headers).  And then, customer can create a an exe with only linking az_iot (without az_core) and they have all functionality from az_core

Would this be fixed using something like???
```c
target_include_directories (az_iot PRIVATE ${CMAKE_SOURCE_DIR}/sdk/core/core/inc)
```

Reply:
I think we should discuss this in the meeting today. I'm not quite sure I understand how this fixes that problem.